runtime/lua/vim/lsp.lua

```diff
@@ -516,6 +516,15 @@ local function lsp_enable_callback(bufnr)
     return
   end
 
+  -- Stop any clients that no longer apply to this buffer.
+  local clients = lsp.get_clients({ bufnr = bufnr, _uninitialized = true })
```

Comment:
I don't know how to justify this `_uninitialized = true` with a test, but I know it's necessary from manual testing with the scenario I described here: https://github.com/neovim/neovim/issues/33443#issuecomment-2800045448